test/unit-tests/function/probability/lgamma.test.js

```diff
@@ -0,0 +1,242 @@
+      CEPSILON
+    )
+    approx.deepEqual(
+      lgamma(math.complex(-2.5, -2.5)),
+      math.complex(-6.27685085148648571668, 6.42804393793984167867),
+      CEPSILON
+    )
+    approx.deepEqual(
+      lgamma(math.complex(-12.5, 12.5)),
+      math.complex(-53.3767553324096369866, -7.23008341760111500158),
+      CEPSILON
+    )
+    approx.deepEqual(
+      lgamma(math.complex(125.5, 125.5)),
+      math.complex(424.211719638964718426, 622.623796700622589246),
+      CEPSILON
+    )
+    approx.deepEqual(
+      lgamma(math.complex(5555.5555, -5555.5555)),
+      math.complex(39906.0402274967320859, -48635.8593112306908421),
+      CEPSILON
+    )
+    approx.deepEqual(
+      lgamma(math.complex(-5000000.5, -5000000.5)),
+      math.complex(-8.56385987850343162145e7, -6.20766433819965678363e7),
+      CEPSILON
+    )
+    approx.deepEqual(
+      lgamma(math.complex(99999999999999.5, 99999999999999.5)),
+      math.complex(3.07973667287988512101e15, 3.23681630555938897728e15),
+      CEPSILON
+    )
+    approx.deepEqual(
+      lgamma(math.complex(-9.87654321e50, 1.23456789e70)),
+      math.complex(-1.93925470679092599791e70, 1.98014414903993314695e72),
+      CEPSILON
+    )
+
+    approx.deepEqual(
+      lgamma(math.complex(0.000123, 123456789)),
+      math.complex(-1.93925479073563285659e8, 2.17671624683482738475e9),
+      CEPSILON
+    )
+    approx.deepEqual(
+      lgamma(math.complex(-0.000123, 123456789)),
+      math.complex(-1.93925479078146610494e8, 2.17671624683444096885e9),
+      CEPSILON
+    )
+    approx.deepEqual(
+      lgamma(math.complex(123456789, -0.000123)),
+      math.complex(2.17671623922326999167e9, -0.00229166241674051621),
+      CEPSILON
+    )
+    approx.deepEqual(
+      lgamma(math.complex(-123456789, -0.000123)),
+      math.complex(-2.17671625601718110732e9, 3.87850942926689860771e8),
+      CEPSILON
+    )
+    approx.deepEqual(
+      lgamma(math.complex(123456789.123, 0.123456789)),
+      math.complex(2.17671624151493240841e9, 2.30017303624303234065),
+      CEPSILON
+    )
+  })
+
+  it('should calculate the lgamma of some special arguments', () => {
+    approx.equal(lgamma(true), 0, EPSILON)
+    assert.strictEqual(lgamma(false), Infinity)
+
+    assert.ok(isNaN(lgamma(NaN)))
+    assert.ok(isNaN(lgamma(-NaN)))
+
+    assert.ok(!isFinite(lgamma(Infinity)))
+    assert.ok(!isFinite(lgamma(-Infinity)))
+  })
+
+  it('should calculate the lgamma of each element in a matrix', () => {
+    approx.deepEqual(
+      lgamma(math.matrix([0, 1, 2, 3, 4, 5])),
+      math.matrix([Infinity, 0, 0, 0.69314718055994530942, 1.79175946922805500081, 3.17805383034794561965])
+    )
+  })
+
+  it('should calculate the lgamma of each element in an array', () => {
+    approx.deepEqual(lgamma([0, 1, 2, 3, 4, 5]), [
+      Infinity,
+      0,
+      0,
+      0.69314718055994530942,
+      1.79175946922805500081,
+      3.17805383034794561965
+    ])
+  })
+
```

Comment:
Of course when you remove the Matrix/Array implementations, you can eliminate these tests.